Lombiq.Analyzers.PowerShell/Rules/Measure-VariableNameFormat/Measure-VariableNameFormat.psm1

```diff
@@ -0,0 +1,344 @@
+                if ([string]::IsNullOrEmpty($nearestParentFunction))
+                {
+                    $nearestParentFunctionName = $rootIndex
+                }
+                else
+                {
+                    $nearestParentFunctionName = $nearestParentFunction.Name
+                }
+
+                # Check if the variable is a parameter, used with or without braces.
+                $matchingParameter = $functionParameterNamesWithParents[$nearestParentFunctionName] | Where-Object {
+                    $PSItem -eq $variableName -or $PSItem -eq $inverseBracedVariableName } | Select-Object -First 1
+
+                # If the variable is not a parameter, check if it starts with a lowercase letter.
+                if ($null -eq $matchingParameter)
+                {
+                    if ($variableName -Match '(?-i)^\$\{?[A-Z].*')
+                    {
+                        $firstLetterIndex = [Math]::Max($variableName.IndexOf('$'), $variableName.IndexOf('{')) + 1
+                        $correctedVariableName = ($variableName.Substring(0, $firstLetterIndex) +
+                            $variableName.Substring($firstLetterIndex, 1).ToLower() +
+                            $variableName.Substring($firstLetterIndex + 1))
+
+                        $correctionExtent = New-Object -TypeName $correctionTypeName -ArgumentList @(
+                            $variable.Extent
+                            $correctedVariableName
+                            "Corrected the casing of the variable name from '$variableName' to '$correctedVariableName'."
+                        )
+
+                        $suggestedCorrections = New-Object System.Collections.ObjectModel.Collection[$correctionTypeName]
+                        $suggestedCorrections.add($correctionExtent) | Out-Null
+
+                        $analyzerViolations += [Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord]@{
+                            'Extent' = $variable.Extent
+                            'Message' = @(
+                                "Variable names should start with a lowercase letter: '$variableName' should be"
+                                "'$correctedVariableName'."
+                            ) -join ' '
+                            'RuleName' = 'PSUseCorrectVariableNameCasing'
+                            'RuleSuppressionID' = 'PSUseCorrectVariableNameCasing'
+                            'Severity' = 'Warning'
+                            'SuggestedCorrections' = $suggestedCorrections
+                        }
+                    }
+
+                    continue
+                }
+                # If a parameter is found, check if it's used exactly as it was declared. The '-ceq' operator should
+                # work here, but it doesn't.
+                elseif (-not $matchingParameter.Equals($variableName, 'InvariantCulture'))
```

Comment:
Is this some kind of PowerShell hack? According to the overload list of `Equals` it shouldn't work, but it magically does.

![image](https://github.com/Lombiq/PowerShell-Analyzers/assets/4007293/811039ef-cc63-41e3-8cf4-e5ced0d4a2f0)

Even so, it would be neater to use the correct .NET overload instead of witchery:
```suggestion
                elseif (-not $matchingParameter.Equals($variableName, [System.StringComparison]::Ordinal))
```

Reply:
It's normal PS syntax regarding enums, but I'll update it to make it explicit.

Reply:
Aha, I didn't realize that `StringComparison.InvariantCulture` is a thing. When I see that name, I assume `CultureInfo.InvariantCulture` which doesn't work for `Equals`.